.buildkite/pipeline.yml

```diff
@@ -394,3 +394,12 @@
     # no TUNE_TESTING=1 on purpose
     - ./ci/travis/install-dependencies.sh
     - bazel test --config=ci $(./scripts/bazel_export_options) --build_tests_only --test_tag_filters=soft_imports python/ray/tune/...
+
+- label: ":steam_locomotive: Train minimal install"
+  conditions: ["RAY_CI_TRAIN_AFFECTED"]
+  commands:
+      - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/travis/upload_build_info.sh; fi }; trap cleanup EXIT
+      - ./ci/travis/install-minimal.sh
+      - ./ci/travis/env_info.sh
+      - python ./ci/travis/check_minimal_install.py
```

Comment:
We may want to consider adding a new `check_train_minimal_install.py` script for the dependencies that we care about.

Theoretically someone could add `torch` install to `install-minimal.sh` and we wouldn't catch that.

But might be out of scope for this PR.

Reply:
Actually nvm the test_minimal.py already checks for this.

Reply:
Yep both these are true. My thought process here is that `install-minimal.sh` will always be the bare minimum for Ray, so if something like `torch` or even `tune` is installed I'd believe there's a bigger problem at hand.

However, the other direction of change would be if `train` has more in its minimal installation path, in which case we should definitely look to use our own script/minimal imports.